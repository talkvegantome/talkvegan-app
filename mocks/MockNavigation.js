export class MockNavigation {
  state = {
    navigationHistory: [],
  };
  addOnNavigateListener() {}
}
