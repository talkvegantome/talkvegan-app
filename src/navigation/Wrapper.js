import React from 'react';
import { Header } from 'react-native-elements';
import { View, SafeAreaView, ScrollView } from 'react-native';
import { commonStyle } from '../styles/Common.style.js';
import { navContainerStyle, navHeaderStyle } from '../styles/Header.style.js'

class Wrapper extends React.Component {
  constructor(props) {
    super(props);
  }
  static defaultProps = {
    style: {}
  }
  render(){
    return (
      <View style={{ flex: 1 }}>
        <Header
          leftComponent={{
            icon: 'menu',
            color: commonStyle.light,
            iconStyle: {paddingLeft: 10},
            onPress: () => this.props.navigation.openDrawer()
          }}
          centerComponent={{
            text: this.props.title,
            style: navHeaderStyle,
          }}
          rightComponent={{
            icon: 'settings',
            color: commonStyle.light,
            iconStyle: {paddingRight: 10},
            onPress: () => this.props.navigation.navigate('Settings')
          }}
          containerStyle={navContainerStyle}
        />
        <SafeAreaView style={{ flex: 1 }}>
          <ScrollView style={{...commonStyle.content,...this.props.style}}>
            {this.props.children}
          </ScrollView>
        </SafeAreaView>
      </View>
    )
  }
}

export default Wrapper
