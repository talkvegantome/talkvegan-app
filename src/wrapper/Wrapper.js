import React from 'react';
import { _ } from 'lodash';
import { View, ScrollView } from 'react-native';
import { commonStyle, PaperTheme } from '../styles/Common.style.js';
import { Provider as PaperProvider, Appbar, BottomNavigation } from 'react-native-paper';

class Wrapper extends React.Component {
  constructor(props){
    super(props)
    this.scrollRef = React.createRef();
  }
  componentDidUpdate(){
    this.scrollRef.current.scrollTo({y: 0, animated: false})
  }
  render(){
    return (
      <View style={{flex: 1}}>
      <PaperProvider theme={PaperTheme} >
        <Appbar.Header style={{backgroundColor: commonStyle.headerBackgroundColor}}>
          <Appbar.Action 
          style={
            // Hide back button if there's no back to go to.
            this.props.navigation.state.navigationHistory.length == 1 ? {display: 'none'} : {}
          }
          icon='keyboard-arrow-left'
          onPress={() => {
            this.props.navigation.goBack()
          }}
          />
          <Appbar.Content
            titleStyle={{
              color: commonStyle.headerFontColor}}
            title={this.props.title}
          />
        </Appbar.Header>
        <ScrollView ref={this.scrollRef} style={{...commonStyle.content,...this.props.style}}>
          {this.props.children}
          <View style={{height:50}}></View>
        </ScrollView>
      </PaperProvider>
      </View>
    ) 
  }
}

export default Wrapper
