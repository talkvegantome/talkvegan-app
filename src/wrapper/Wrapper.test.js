import React from 'react';
import Wrapper from './Wrapper.js';
import renderer from 'react-test-renderer';

import {MockNavigation} from '../../mocks/MockNavigation.js'
let mockNavigation = new MockNavigation()


test('Wrapper Renders Correctly', () => {
  const tree = renderer.create(<Wrapper
    navigation={mockNavigation}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
