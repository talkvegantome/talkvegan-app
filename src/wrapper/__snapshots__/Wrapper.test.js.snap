// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Renders Correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "backgroundColor": "#143642",
            "zIndex": 0,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#143642",
            "elevation": 0,
            "flexDirection": "row",
            "height": 56,
            "marginTop": 0,
            "paddingHorizontal": 4,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          clickable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "height": 36,
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "visible",
                  "width": 36,
                },
                undefined,
                Object {
                  "display": "none",
                },
              ],
            ]
          }
          testID="back_button"
        >
          <View
            style={null}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  îŒ”
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          clickable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              Array [
                Object {
                  "marginLeft": 8,
                },
                Object {
                  "alignItems": "center",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            accessibilityRole="header"
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                ],
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
      </View>
    </RCTSafeAreaView>
    <RCTScrollView
      onScroll={[Function]}
      scrollEventThrottle={0}
      style={
        Object {
          "paddingBottom": 20,
          "paddingLeft": 30,
          "paddingRight": 30,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "height": 50,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
